src/ui/Ui.java

```diff
@@ -0,0 +1,184 @@
+     * @return A string message indicating the task has been added.
+     */
+    public String getTaskAdded(Task task, int taskCount) {
+        return String.format("Got it. I've added this task:\n  %s (priority: %d)\nNow you have %d tasks in the list.",
+                task.toString(), task.getPriority(), taskCount);
+    }
+
+    /**
+     * Returns a message indicating that a task has been deleted.
+     *
+     * @param task The task that has been deleted.
+     * @param size The current number of tasks in the list.
+     * @return A string message indicating the task has been deleted.
+     */
+    public String getTaskDeleted(Task task, int size) {
+        return "Noted. I've removed this task:\n[Priority: "
+                + task.getPriority() + "] " + task + "\nNow you have " + size + " tasks in the list.";
+    }
+
+    /**
+     * Returns an error message.
+     *
+     * @param message The error message to be displayed.
+     * @return A string containing the error message.
+     */
+    public String getError(String message) {
+        return "Error: " + message;
+    }
+
+    /**
+     * Returns a goodbye message.
+     *
+     * @return A string containing the goodbye message.
+     */
+    public String getGoodbye() {
+        return "Bye. he he he";
+    }
+
+    /**
+     * Returns a list of tasks that match the given argument.
+     *
+     * @param argument The argument to match against task descriptions.
+     * @param tasks The list of tasks to search.
+     * @return A string containing the list of matching tasks.
+     */
+    public String getMatchingTasks(String argument, Task... tasks) {
+        StringBuilder sb = new StringBuilder("Here are the matching tasks in your list:\n\n");
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy");
+        LocalDate inputDate = null;
+        int count = 1;
+
+        // Try to parse the argument as a date
+        try {
+            inputDate = LocalDate.parse(argument, formatter);
+        } catch (DateTimeParseException e) {
+            // If parsing fails, treat the argument as a normal string
+        }
+
+        for (Task task : tasks) {
+            if (inputDate != null) {
+                if (task instanceof Deadline deadline) {
+                    if (deadline.getBy().toLocalDate().isEqual(inputDate)) {
+                        sb.append(count).append(") [Priority: ").append(
+                                task.getPriority()).append("] ").append(task).append("\n\n");
+                        count++;
+                    }
+                } else if (task instanceof Event event) {
+                    LocalDate startDate = event.getFrom().toLocalDate();
+                    LocalDate endDate = event.getTo().toLocalDate();
+                    if ((inputDate.isEqual(startDate) || inputDate.isAfter(startDate))
+                            && (inputDate.isEqual(endDate) || inputDate.isBefore(endDate))) {
```

Comment:
Is it possible to split this into individual boolean values?
(eg. isGTEStartDate and isLTEEndDate)